profiler/src/ProfilerEngine/Datadog.Profiler.Native/CorProfilerCallback.cpp

```diff
@@ -179,7 +180,8 @@ bool CorProfilerCallback::InitializeServices()
             _pThreadsCpuManager,
             _pAppDomainStore.get(),
             pRuntimeIdStore,
-            _metricsRegistry);
+            _metricsRegistry,
+            CallstackPoolManager::GetDefault());
```

Comment:
Is it expected that we use the same CallStack for exceptions, allocations, and contentions?
Each provider could be called from a different thread (i.e. our memory resource implementation used to "get" a callstack should be thread safe). Also, the dimension of the pool could vary per provider. Last but not least, having one pool per provider will help to identify each one is consuming memory the most

Reply:
you are mixing provider/pool. In the current implementation, there is no provider.
A pool manager returns a pool. The default pool will use a default allocator (new/delete) and there is no need to lock or create different instances. The locking mechanism is done by the low level allocator.